### ISSUE TITLE
训练以及测试时为什么drop_last=True?

### ISSUE THREAD

**Issue by Sunting78** [thuml/Nonstationary_Transformers#26]
这样训练会少很多样本；
测试时也会缺少样本，导致指标不正确。
论文的指标也是这样的setting下得到的吗？

比如 ill数据集 36in 36pred
Namespace(activation='gelu', batch_size=32, c_out=7, checkpoints='./checkpoints/', d_ff=2048, d_layers=1, d_model=512, data='custom', data_path='national_illness.csv', dec_in=7, des='Exp_h32_l2', devices='0,1,2,3', distil=True, do_predict=False, dropout=0.05, e_layers=2, embed='timeF', enc_in=7, factor=3, features='M', freq='h', gpu=1, is_training=1, itr=1, label_len=18, learning_rate=0.0001, loss='mse', lradj='type1', model='ns_Transformer', model_id='ili_36_36', moving_avg=25, n_heads=8, num_workers=10, output_attention=False, p_hidden_dims=[32, 32], p_hidden_layers=2, patience=3, pred_len=36, root_path='./illness/', seed=2021, seq_len=36, target='OT', train_epochs=10, use_amp=False, use_gpu=True, use_multi_gpu=False)

测试时drop_last=False:
test shape: (158, 36, 7) (158, 36, 7)  
mse:2.9380314350128174, mae:1.051339030265808
测试时drop_last=True:
test shape: (128, 36, 7) (128, 36, 7)  --- 缺少了158-128个测试样本
1.8928638696670532, mae:0.8517037034034729

麻烦验证下。

**Comment by WenWeiTHU** [comment]
Thanks for pointing out this and it is a historical problem since this repo is built from Autoformer: https://github.com/thuml/Autoformer/issues/104.

We will fix the problem reevaluated the results soon.
